src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -1,360 +1,231 @@
       }
     }
   }
 
-  async function approveWithdraw(posManager: Contract | null, contractAddress: string) {
+  async function approveWithdraw() {
+    if (!provider) {
+      notifyError('Please connect wallet')
+      return
+    }
+
     if (!accountInfo || !accountInfo.address) {
-      notifyError('Wallet not connected.')
+      notifyError('User wallet not found.')
       return
     }
-    if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-      if (posManager === null) {
-        return null
+
+    try {
+      if (parseInt(totalLiquidityAmt) === 0) {
+        const error: any = new Error('User does not have reserves to withdraw')
+        error.code = '4002'
+        throw error
+      }
+      if (!gammaPoolContract) {
+        await initializeGammaPoolContract()
+      }
+      if (provider && gammaPoolContract) {
+        let tx = await gammaPoolContract.approve(POSITION_MANAGER_ADDRESS, constants.MaxUint256.toString())
+        return await tx.wait()
       } else {
-        if (provider !== null) {
-          try {
-            let address = process.env.NEXT_PUBLIC_GAMMAPOOL_ADDRESS
-            let gammaPoolContract
-            if (provider && address) {
-              if (accountInfo && accountInfo?.address) {
-                gammaPoolContract = new ethers.Contract(
-                  address,
-                  GammaPool.abi,
-                  provider.getSigner(accountInfo?.address)
-                )
-              } else {
-                gammaPoolContract = new ethers.Contract(address, GammaPool.abi, provider)
-              }
-              let tx = await gammaPoolContract.approve(contractAddress, constants.MaxUint256.toString())
-              return await tx.wait()
-            } else {
-              notifyError('Please connect wallet')
-            }
-          } catch (e) {
-            throw e
-          }
-        } else {
-          notifyError('Please connect wallet')
-        }
+        notifyError('Please connect wallet')
       }
-    } else {
-      if (depPool === null) {
-        return null
+    } catch (e: any) {
+      if (e?.code && e?.code === '4002') {
+        throw new Error(e.message)
       } else {
-        if (provider !== null) {
-          try {
-            let tx = await depPool.approve(contractAddress, constants.MaxUint256.toString())
-            return await tx.wait()
-          } catch (e) {
-            throw e
-          }
-        } else {
-          notifyError('Please connect wallet')
-        }
+        throw new Error('An error occurred while approval')
       }
     }
   }
 
   useEffect(() => {
-    async function fetchContract() {
-      if (!provider) {
-        notifyError('Please connect wallet.')
+    async function fetchData() {
+      if (!gammaPoolContract) {
         return
       }
+      const liqBal = await gammaPoolContract.balanceOf(accountInfo?.address)
+      console.log(liqBal.toString())
+      setEnableApprove(liqBal.toString() > 0)
```

Comment:
How can I get USER_POSITION_ID.

We can also do like what Godwin did for adding liquidity. Like positions/deposit/<gammapooladdress>

Reply:
Separated the file which will accept the pool's address in url and will act upon it

Reply:
Theres' no position ID. Ther'es only the user's wallet if he has any GSP Tokens or not. So the positions page should look at the user's wallet to see if he has any GSP Tokens balances. If so, They can go into the withdraw page for the pool they have GSP tokens in. In this case the withdraw should be prepopulated with the user's GSP balance. If there's no balance, then the page will be empty and not populated.

Reply:
Can you fix that getBalance shows an error?

Reply:
>Theres' no position ID for withdrawal
he's right. it would just have the `poolId` of the pool that the user wants to withdraw from prepopulated with his balance